data/test/scenarios/events-test_pre_attack.cfg

```diff
@@ -0,0 +1,194 @@
+# Check that modifications to the attack being used are applied to the combat
+##
+# Expected end state:
+# All automatically stored variables are present.
+# The Elvish Archer's sword attack is modified to have 1 damage, resulting in the Orcish Grunt having 36/38 hitpoints.
+#####
+{GENERIC_UNIT_TEST "events-test_pre_attack_change_weapon" (
+    [event]
+        name=start
+        [do_command]
+            [move]
+                x=7,8,9,10,11,12
+                y=3,3,3,3,3,3
+            [/move]
+        [/do_command]
+        [do_command]
+            [attack]
+                weapon=0
+                defender_weapon=0
+                [source]
+                    x,y=12,3
+                [/source]
+                [destination]
+                    x,y=13,3
+                [/destination]
+            [/attack]
+        [/do_command]
+    [/event]
+    [event]
+        name=pre_attack
+        # TODO: change selected attack, test result of combat with modified attack
+        # test unit and second_unit
+        [if]
+            [not]
+                [variable]
+                    name=unit.type
+                    equals="Elvish Archer"
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        [if]
+            [not]
+                [variable]
+                    name=second_unit.type
+                    equals="Orcish Grunt"
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        # test weapon and second_weapon
+        [if]
+            [not]
+                [variable]
+                    name=weapon.name
+                    equals="sword"
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        [if]
+            [not]
+                [variable]
+                    name=second_weapon.name
+                    equals="sword"
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        # test x1,y1
+        [if]
+            [not]
+                [variable]
+                    name=x1
+                    equals=12
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        [if]
+            [not]
+                [variable]
+                    name=y1
+                    equals=3
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        # test x2,y2
+        [if]
+            [not]
+                [variable]
+                    name=x2
+                    equals=13
+                [/variable]
+            [/not]
+            [then]
+                {FAIL}
+            [/then]
+        [/if]
+        [if]
```

Comment:
Each of these if statements should be reduced to an `{ASSERT}`.

Reply:
Updated.